splunk_eventgen/eventgen_core.py

```diff
@@ -234,22 +238,6 @@ def _create_generator_pool(self, workercount=20):
                     worker.setDaemon(True)
                     worker.start()
 
-    def _create_generator_workers(self, workercount=20):
```

Comment:
The original intent of this other structure was to allow the main eventgen program to receive config updates, change and dynamically recreate and allocate the pools.  The new style here will lock on the join's and will then wait for the pool.  Are we sure we want to go that route?  If you want to change config, you have to restart eventgen?  If we do, I think this is fine, however the "max_tasks" fix is really more about the queues getting backed up and filling up due to eventgen not being fast enough, and I think this is overkill for the fix.

Reply:
For `multiprocess` mode, the config will only load once after starting EventGen. (Correct me if I am wrong)

And by leveraging `max_tasks`, we leave the memory management task to the process pool. I have also tried other ways such as getting the queue larger, explicitly delete large object after usage which are not working to fix the OOM issue. It also takes long time to verify each try and test. Maybe this is not the best solution but it solves the issue for us to release the last version of Py2.

Reply:
Yeah the original method that was created was to have the config object be able to have a "max workers" and to be able to update the worker count and then recreate all the pools and workers dynamical.  I'm fine if we kill off that logic, that's more of a lets make the call in our next meeting

Reply:
> the original method that was created was to have the config object be able to have a "max workers" and to be able to update the worker count and then recreate all the pools and workers dynamical.
This is the initial thought right? The currently logic is not what you said above. 

Then let's ship this and discuss it on our next meeting.